macroslib/src/java_jni/mod.rs

```diff
@@ -89,16 +89,11 @@ impl ForeignMethodSignature for JniForeignMethodSignature {
 }
 
 impl LanguageGenerator for JavaConfig {
-    fn generate(
```

Comment:
I added test case for your PR: b3f19f67151d7e977811900c642f8a8cc2420cf7 

and it **fails** for Java case:

> Can not find such rust type '& B' Can not find conversation from 'jlong' to '& B'

The simple approach I suppose to force it to work correctly:
-  it should parse each classes, but skip methods, and deal only with constructors, and then re-parse again with methods.

But of course if constructors accept class that not yet defined then this is fatal error,
the more robust and complex solution is
- parse only constructors, deals only with return values from constructors to maker familiar with the way how to create classes, and then repeat the same thing without ignoring method and input parameters of constructors

or may be 
- allow forward declarations, like:
```Rust
foreigner_class!(class A {
    self_type A;
    constructor A::default() -> A;  
});

foreigner_class!(class B {
    self_type B;
    constructor B::default() -> B;
    static_method B::b(a: &A);
});

foreigner_class!(class A {
    self_type A;
    constructor A::default() -> A;
    static_method A::a(b: &B);
});

```

Reply:
Oh, I didn't notice registration code in `generate_rust_code` function. I've moved it into `register_class`, just like I did with C++ and now the test works.

Let me know if this solution is OK and if I should squash all changes into two commits, like previously.